### ISSUE TITLE
Docker compose networks works incorrect (WSL and Hyper-V) after 4.21.1 update

### ISSUE THREAD

**Issue by mrFleshka** [docker/for-win#13610]
### Description

After docker compose up containers randomly do not added to networks (default or external).
For example i create [test project](https://github.com/mrFleshka/test-docker-networking) with 5 containers in compose file.
Every container have 3 networks.
On screenshots you can see that they attachet not all time.
![TestDockerNetworks – README md 2023-07-19 22 49 39](https://github.com/docker/for-win/assets/14197648/b7015781-6492-4f22-85a3-4902e7d59ad7)
![TestDockerNetworks – README md 2023-07-19 22 50 02](https://github.com/docker/for-win/assets/14197648/50456f78-f7c5-4763-b576-65b2fc1696ce)
![TestDockerNetworks – README md 2023-07-19 22 50 12](https://github.com/docker/for-win/assets/14197648/c5881faf-50c6-4276-baf4-592157f4df24)
![TestDockerNetworks – README md 2023-07-19 22 50 30](https://github.com/docker/for-win/assets/14197648/b4a00a42-4994-429f-ab50-47f918188ff8)

### Reproduce

1. Clone project or copy project files [from github](https://github.com/mrFleshka/test-docker-networking)
2. Run `docker compose up -d` in directory
3. Check all containers for all proxy-pass networks (i check it via phpstorm services docker panel)
4. Down all containers `docker compose down ` and repeat while not get bug (not 5 containers in network)

### Expected behavior

Every network must have all 5 containers all time (for this test project)

### docker version

```bash
Client:
 Cloud integration: v1.0.35
 Version:           24.0.2
 API version:       1.43
 Go version:        go1.20.4
 Git commit:        cb74dfc
 Built:             Thu May 25 21:53:15 2023
 OS/Arch:           windows/amd64
 Context:           default

Server: Docker Desktop 4.21.1 (114176)
 Engine:
  Version:          24.0.2
  API version:      1.43 (minimum version 1.12)
  Go version:       go1.20.4
  Git commit:       659604f
  Built:            Thu May 25 21:52:17 2023
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.21
  GitCommit:        3dce8eb055cbb6872793272b4f20ed16117344f8
 runc:
  Version:          1.1.7
  GitCommit:        v1.1.7-0-g860f061
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Version:    24.0.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.19.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.6
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.16.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 61
 Server Version: 24.0.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 3dce8eb055cbb6872793272b4f20ed16117344f8
 runc version: v1.1.7-0-g860f061
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 5.15.49-linuxkit-pr
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.938GiB
 Name: docker-desktop
 ID: aae330db-c04a-428c-a4b0-4d9b5aad1cae
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 48
  Goroutines: 67
  System Time: 2023-07-19T20:12:05.4547588Z
  EventsListeners: 9
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false
```

### Diagnostics ID

4474A536-E3A5-4F4F-82F0-2EC84777C9B5/20230719201404

### Additional Info

No idea why it does. I check it in Hyper-V and WSL.
Also check it on Win 10 and Win 11 and in version 4.21.1 its broken =(

**Comment by lorenrh** [comment]
Thanks for your report!

Unfortunately you have stumbled on one of our known issues in compose (more info on the [compose release notes](https://github.com/docker/compose/releases/tag/v2.19.1)).

But there's a fix to be included in the next release of Docker Desktop! Sorry for the inconvenience. 🙇

**Comment by mrFleshka** [comment]
All fixed in Docker Desktop v4.22.0
